apps/dot-voting/app/components/VoteDetails/index.js

```diff
@@ -54,8 +61,28 @@ const VoteDetails = ({ app, vote, userAccount, onVote }) => {
       }
     }
 
+    function loadUserBalance() {
+      const { snapshotBlock } = vote.data
+      if (tokenContract && userAccount) {
+        combineLatest(
+          tokenContract.balanceOfAt(userAccount, snapshotBlock),
+          tokenContract.decimals(),
+          tokenContract.symbol()
+        )
+          .pipe(first())
+          .subscribe(([ balance, decimals, symbol ]) => {
+            setTokenData({
+              userBalance: balance,
```

Comment:
I don't see this, either. The function is called `loadUserBalance`, as if it's the most important attribute returned, but then we only ever use `tokenData.decimals`. I think we can simplify all of this logic a bit, and store `decimals` directly in state, rather than storing an object for `tokenData`.

Reply:
Yes, balance is sourced directly from the contract in store... simplified.